master/internal/provisioner/aws_config.go

```diff
@@ -215,3 +195,106 @@ func (t ec2InstanceType) Validate() []error {
 		errors.Errorf("ec2 instance type must be valid type: %s", strings.Join(strs, ", ")),
 	}
 }
+
+// This map tracks how many slots are available in each instance type. It also
+// serves as the list of instance types that the provisioner may provision - if
+// the master.yaml is configured with an instance type not on this list, the
+// provisioner will consider it an error.
+var ec2InstanceSlots = map[ec2InstanceType]int{
+	"g4dn.xlarge":   1,
+	"g4dn.2xlarge":  1,
+	"g4dn.4xlarge":  1,
+	"g4dn.8xlarge":  1,
+	"g4dn.16xlarge": 1,
+	"g4dn.12xlarge": 4,
+	"g4dn.metal":    8,
+	"p2.xlarge":     1,
+	"p2.8xlarge":    8,
+	"p2.16xlarge":   16,
+	"p3.2xlarge":    1,
+	"p3.8xlarge":    4,
+	"p3.16xlarge":   8,
+	"p3dn.24xlarge": 8,
+	"t2.medium":     0,
+	"t2.large":      0,
+	"t2.xlarge":     0,
+	"t2.2xlarge":    0,
+	"t3.nano":       0,
```

Comment:
wondering if this instance type actually is able to run docker.

Reply:
It is